plat/xen: Remove __i386__ code

Sairajkodilkar opened the pull request:
The unikraft is never going to support the 32-bit on x86, hence this
commit removes the redundant code introduced due to "#if defined(\_\_i386__) and
Closes #26

<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.


### Base target

 - Architecture(s): x86_64
 - Platform(s):  xen 
 - Application(s): N/A


### Additional configuration

<!--
Please specify any additional configuration which is needed for this feature to
work or any new configuration parameters which are introduced by this PR.  This
will help during the review process.  For example:

 - `CONFIG_LIBUKDEBUG=y`

-->

### Description of changes
As mentioned in issue #26 Unikraft is never going to support the i386 architecture, hence the commit removes the redundant code introduced by #if defined(\_\_i386__).


<!--
Please provide a detailed description of the changes made in this new PR.
-->

cristian-vijelie reviewed (approved):
It looks fine, thank you!

Reviewed-by: Cristian Vijelie <cristianvijelie@gmail.com>

nderjung reviewed (changes requested):
Hi, thanks for this PR.

Please can you address the commit length.  My suggestion is:

```
plat/xen: Remove `__i386__` code

Unikraft is never going to support the i386 platform hence this commit
removes the redundant code introduced by "if defined(__i386__)"

GitHub-Fixes: #26
Signed-off-by: Sairaj Kodilkar <skodilkar7@gmail.com>
```

Also cc'ing @unikraft/sig-plat-xen for additional confirmation on this PR.

Sairajkodilkar commented:
hi @nderjung, I am trying to make the changes using git commit --amend but facing issues due to branch divergence since the changes are already pushed, can you please help me with this issue?

cristian-vijelie commented:
@Sairajkodilkar you can use `push --force`

Sairajkodilkar commented:
hi @cristian-vijelie thanks for the help, I have done the necessary changes

skuenzer commented on plat/xen/include/xen/arch-x86/xen.h:
```diff
@@ -56,19 +56,7 @@
-__DeFiNe__ __DECL_REG_LO8(which) uint32_t e ## which ## x
-__DeFiNe__ __DECL_REG_LO16(name) union { uint32_t e ## name; }
-# endif
-#include "xen-x86_32.h"
```
I think the included file should be removed, too.

skuenzer commented on plat/xen/include/xen/arch-x86/xen.h:
```diff
@@ -193,26 +181,6 @@ struct vcpu_guest_context {
-    unsigned long failsafe_callback_cs;     /* CS:EIP of failsafe callback  */
-    unsigned long failsafe_callback_eip;
-#else
-    unsigned long event_callback_eip;
```
This is in an else case for __i386__, so everything else to x86. It should stay, right?

Sairajkodilkar replied:
yes, it should stay. sorry for the mistake, I accidentally removed it.

skuenzer commented on plat/xen/include/xen/io/protocols.h:
```diff
@@ -29,9 +29,7 @@
 #define XEN_IO_PROTO_ABI_X86_64     "x86_64-abi"
 #define XEN_IO_PROTO_ABI_ARM        "arm-abi"
 
-#if defined(__i386__)
```
The headers within `plat/xen/inckude/xen` should not be touched. These are a copy from upstream Xen that are updated/synced with Xen.

Sairajkodilkar replied:
Thanks for the review, I will restore these headers in the next commit.

Sairajkodilkar replied:
Hey, Does this mean that I should only change the headers within the file xen-x86/mm.h since this is the only file that contains __i386__ headers and is not under the directory plat/xen/include/xen.

skuenzer reviewed (changes requested):
Hey @Sairajkodilkar ,
thanks for cleaning up Xen code for x86. I commented some things that I found while scanning the PR. I would also try to remove headers that aren't  used anymore. The only folder we should not touch is `plat/xen/include/xen`. In there is a copy of official Xen headers that are synced with upstream Xen.
Thanks,
Simon

Sairajkodilkar committed:
plat/xen: Remove __i386__ code

Unikraft is never going to support the i386 platform hence this commit
removes the redundant code introduced by "if defined(__i386__)". This
commit does not remove the __i386__ code from the directory
plat/xen/include/xen since it is synced with the upstream Xen.

GitHub-Fixes: #26
Signed-off-by: Sairaj Kodilkar <skodilkar7@gmail.com>

Sairajkodilkar commented:
Hey @skuenzer I have restored the changes done under the xen folder.

skuenzer reviewed (approved):
Thanks a lot!

Reviewed-by: Simon Kuenzer <simon.kuenzer@neclab.eu>
Approved-by: Simon Kuenzer <simon.kuenzer@neclab.eu>
